### DISCUSSION TITLE
Here's my current Shape for Resources

### DISCUSSION THREAD

**Discussion by YetAnotherJonWilson** [solid-contrib/getting-started#22] [Ideas]
UPDATE 12/8: per the discussion in Matrix chat, I think what I'm trying to define here is a Shape--I previously used the term ontology, but that apparently is the broader vocabulary from which a shape is derived.

(This is a **work in progress**, I need to figure out how to formally structure a shape.)

documentation resource
 schema:name <string>
 schema:about <string>[]
 schema:category <string>
    can be one of the following:
	 - Reference
	 - Tutorial
	 - How-to
	 - Explanation
 schema:url <URL>
 (schema:type or similar) <string>[]
	 [video, book, library, app, etc.]
 (schema:createdOn or similar) <datetime>
 (schema:createdBy or similar) <string>
 (schema:updatedOn or similar) <datetime>[]
 (schema:updatedBy or similar) <string>[]
 (most recent datetime checked for up-to-date status) <string>
 (up-to-date status) <string>

clarifying examples regarding tutorials:
    an app that is included in the resources is intended to serve as an example for developers to learn from (whatever the intention of the app maker), and, because it is an app, and not a more specific how-to, it should therefore be categorized as a tutorial (even if it is not presented as a tutorial) **because the purpose of the app as a documentation resource is the same as the purpose of a tutorial**--to demonstrate for developers how to build an application

Principles:
 When categorizing resources, think about **the purpose** that the resource serves as a form of documentation: is it more useful while actively building an app (how-to's and reference), or when learning more generally (tutorials and explanation)?; is it task-oriented (how-to)?; is it a broad topic, or several topics, demonstrating, for example, an entire app (tutorial)?; is it used to get an answer to a specific question while building an app (reference), or to try to gain a better understanding of a topic (explanation)?

**Comment by danielbakas** [comment]
Hi @YetAnotherJonWilson! This is a **great** idea. Defining an Ontology for Solid Resources (OSR maybe?) could be absolutely valuable.

Some thoughts:
1. We could first define a **vocabulary** for the resources and attributes we're most interested in representing. A "Vocabulary for Solid Resources" would allow us to define what we mean by every thing. Then we could construct an Ontology based on that Vocabulary 🤓
1. Going for schema.org is a solid (no pun intended) choice, but it might be subjective and limited in some areas.
1. The clear division of metadata and data about the resource will be crucial. 
1. At Semantyk, we currently have "Audiovisual, Readable, Organization, Project, Process, Person..." and many other types of "resources" related to Solid. What would be the scope of **this** Knowledge Graph?
1. I believe a form of some sort would be really useful. We'd love to contribute if this is something interesting or worth pursuing

We hope these thoughts are valuable. We'd love to know everyone's take on them.

Also, we (at Semantyk) have some (little but useful) experience with ontology engineering, and once we defined which attributes or types we'd be most interested in, we'd love to contribute in designing and drafting the vocab/ontology.

Excited to know what you think 🤓

**Comment by YetAnotherJonWilson** [comment]
Hi Daniel, I updated the discussion title based on our chat on Matrix. My focus, for now (and I hope I'm using these terms correctly) is about defining a shape to be used for creating the resources. The shape will determine the fields available in the form, and will be used to submit the request to create the new resource in the pod.

Having said that, I think it's a great idea to be working on a broader vocabulary (or ontology, perhaps) for Solid resources. It sounds like you've already begun to make some progress in that area, and that will probably be very useful to us in the future. Also, I think Jeff mentioned that this is something he's been working a lot on, and will be releasing something soon. So I think there will be some good Matrix chats happening about ontologies and resources.

Please feel free to respond here, if you'd like, especially if you have feedback about defining shapes and using shapes generally, or about this specific shape I'm working on. Or we could also discuss vocabularies and ontologies more broadly over in the Matrix chat.

I do appreciate your enthusiasm and contributions to this and other discussions--I really hope that I can figure out how to take advantage of your knowledge and willingness to help! Thank you!

**Comment by danielbakas** [comment]
You are very kind @YetAnotherJonWilson, and like you say, you might be giving me more credit than I deserve 😅 I am glad though that our discussions have been productive and exciting so far 😄

I think we mentioned it over at the Matrix chat, but just to mention it over here, studying closely what @jeff-zucker is proposing with the [Shapes for Products and Services #7](https://github.com/solid-contrib/practitioners/discussions/7) might be really valuable and related to what we're trying to accomplish here.

I 100% agree with you that our focus is defining a shape for our resources. And I really like @jeff-zucker's approach of defining or identifying as many shared predicates between all our Solid Resources as possible, since this would allow us to move forward from a common ground of what all our Solid Resources look like (and @jeff-zucker may correct me if I'm making no sense at all 😂).

So that's where I would put my energy on, right now. On answering the question "What do all Solid resources have in common?" and "How do we represent these shared attributes using predicates from existing vocabularies and ontologies?" since that too will make it simpler by reducing the need to define our own vocabularies.

What do you think? 🤓
